### PR TITLE
Improve browsing companies and jobs

### LINKED ISSUE #17: Creating company while logged out generates error
Working through #12 and running into an issue.  Not sure if this is just because I haven't setup the github integration yet but as a logged out user when trying to create a new company I am getting the following error:

no function clause matching in Companies.PendingChanges.insert_change/4

The last param for insert_change is nil when logged out when the only function def requires a map.

To recreate log out and try to add a company.

### PR DESCRIPTION
**Pull request by doomspork**
Closes #17 and #13. I think #14 should be handled in a separate PR with more thought given to how we store location (we probably want to store lat/lon; truely remote companies though?).

Not ready just yet. Still need to do some more work.

### CODE REVIEW

File: lib/companies_web/helpers/user_helpers.ex [hunk 1]
```diff
@@ -9,4 +9,12 @@ defmodule CompaniesWeb.UserHelpers do
   end
 
   def current_user(conn), do: conn.assigns[:current_user]
+
+  def signed_in?(conn) do
+    with %{id: _id} <- Map.get(conn.assigns, :current_user) do
```
Comment (doomspork) [inline lib/companies_web/helpers/user_helpers.ex]:
```elixir
def signed_in?(conn), do: current_user(conn) != nil
```

**Comment by doomspork** [comment]
1. Focus on job listings, not companies hiring. I think we should emphasize the job listings more. Make them easier to access here, not footnotes on company cards.
1. Based on all the feedback we get searching by industry is not useful as browsing might be. This should allow folks to browse by name at the very least.
1. When signed out, call-to-action to sign-in and list. I'm not 💯 on this. I kind of think we need a dedicated sign-in screen that we can redirect to.

image

**Comment by doomspork** [comment]
image

@gemantzu ☝️

File: lib/companies/repo.ex [hunk 2]
```diff
@@ -3,6 +3,8 @@ defmodule Companies.Repo do
     otp_app: :companies,
     adapter: Ecto.Adapters.Postgres
 
+  use Scrivener, page_size: 20
```
Comment (doomspork) [inline lib/companies/repo.ex]: I set the default elsewhere to `8`, let's change this

File: lib/companies_web/controllers/company_controller.ex [hunk 3]
```diff
@@ -4,27 +4,21 @@ defmodule CompaniesWeb.CompanyController do
   alias Companies.{Companies, Industries, Schema.Company}
 
   def recent(conn, _params) do
-    companies_count = Companies.count_total()
-    recent_companies = Companies.list_recent_companies()
+    companies_count = Companies.count()
+    recent_companies = Companies.all(%{})
```
Comment (doomspork) [inline lib/companies_web/controllers/company_controller.ex]: Let's add a default value to `all/1` to avoid this need for an empty map.

**Comment by doomspork** [comment]
image

This is the only outstanding issue. The query for hiring companies is a little wonky and so it pages even weirder.

**Comment by gemantzu** [comment]
Please test this, both company query and job creation, I thnk we are good to go live!

**Pull request merged**